### ISSUE TITLE
ReduceLROnPlateau mode be faithful to the validation metric passed

### ISSUE THREAD

**Issue by codedecde** [allenai/allennlp#2895]
**Is your feature request related to a problem? Please describe.**
Currently, for the ReduceLROnPlateau (or any class subclassing _PyTorchLearningRateSchedulerWithMetricsWrapper), the mode (whether to take a step when a metric increases or decreases) has to be specified independent of the validation metric ("-loss" / "+acc" etc.). This can potentially lead to bugs with scheduling. Eg: the pytorch default for ReduceLROnPleateau uses a min for scheduling. So if you want to schedule based on accuracy, that leads to a bad schedule, if not specified in the learning rate scheduler.

**Describe the solution you'd like**
Make sure that the from_params of the trainer specifies the mode for a learning rate scheduler if not specified, that is faithful to the "+/-" value of the validation metric, if specified. If the mode is not specified, then set it appropriately. 
Somewhere after
https://github.com/allenai/allennlp/blob/d6bb6c87581bc3a37ec7cccbb1778364344b1a7f/allennlp/training/trainer.py#L674
would be nice.
Optionally, a check in the \_\_init\_\_() function would also be nice, to make sure someone not using the from_params structure gets warned if the mode and the metric are not faithful to each other.

**Describe alternatives you've considered**
Another alternative would be that __PyTorchLearningRateSchedulerWrapper requires the mode param to be specified, i.e the default is never used.

**Comment by DeNeutoy** [comment]
This would be a good PR - would you mind writing it? we use the `ConfigurationError` for raising errors which arise from mis-specified configs (`allennlp.common.checks.ConfigurationError`).

**Comment by codedecde** [comment]
Sure will do.
Just wanted to make sure what the would be the desired behaviour before.
Seems like making _PyTorchLearningRateSchedulerWrapper requiring the mode to be specified, and updating the Trainer.from_params to add it to the scheduler params if not specified would be sufficient, since then we are certain that the mode is always intentionally specified, or is faithful to the validation metric.
